### PR TITLE
As a user i want to upgrade the RocksDB to version 5.17.2 to be able to benefit from bug fixes and performance improvements

### PR DESCRIPTION
**Pull request by legacycode**
# Description

Actually i am not sure, how this project manages the product life cycle. Since RocksDB 5.7.3 a lot of bug fixes and performance improvements were implemented. More details at:

https://github.com/facebook/rocksdb/releases

# How Has This Been Tested?

I upgraded RocksDB step by step and just started IRI. This was not really tested on my side.

### CODE REVIEW

**Comment by GalRogozinski** [comment]
Updating RocksDb is on our roadmap

When we get around to doing the proper tests we will merge

**Comment by GalRogozinski** [comment]
This will be merged after the JMH RocksDbBenchMark runs

**Comment by GalRogozinski** [comment]
```Benchmark                         (numTxsToTest)  Mode  Cnt   Score    Error  Units
RocksDbBenchmark.deleteBatch                  10  avgt   10   1.004 ?  0.055  ms/op
RocksDbBenchmark.deleteBatch                 100  avgt   10   1.689 ?  0.129  ms/op
RocksDbBenchmark.deleteBatch                 500  avgt   10   4.227 ?  0.447  ms/op
RocksDbBenchmark.deleteBatch                1000  avgt   10   7.804 ?  0.418  ms/op
RocksDbBenchmark.deleteBatch                3000  avgt   10  21.852 ?  2.694  ms/op
RocksDbBenchmark.deleteOneByOne               10  avgt   10   0.042 ?  0.003  ms/op
RocksDbBenchmark.deleteOneByOne              100  avgt   10   0.485 ?  0.099  ms/op
RocksDbBenchmark.deleteOneByOne              500  avgt   10   2.372 ?  0.138  ms/op
RocksDbBenchmark.deleteOneByOne             1000  avgt   10   5.284 ?  0.873  ms/op
RocksDbBenchmark.deleteOneByOne             3000  avgt   10  17.131 ?  2.191  ms/op
RocksDbBenchmark.dropAll                      10  avgt   10   0.147 ?  0.038  ms/op
RocksDbBenchmark.dropAll                     100  avgt   10   1.137 ?  0.219  ms/op
RocksDbBenchmark.dropAll                     500  avgt   10   6.521 ?  2.547  ms/op
RocksDbBenchmark.dropAll                    1000  avgt   10  13.225 ?  4.656  ms/op
RocksDbBenchmark.dropAll                    3000  avgt   10  24.920 ?  7.459  ms/op
RocksDbBenchmark.fetchOneByOne                10  avgt   10   0.043 ?  0.004  ms/op
RocksDbBenchmark.fetchOneByOne               100  avgt   10   0.585 ?  0.398  ms/op
RocksDbBenchmark.fetchOneByOne               500  avgt   10   4.857 ?  3.826  ms/op
RocksDbBenchmark.fetchOneByOne              1000  avgt   10  14.119 ?  6.334  ms/op
RocksDbBenchmark.fetchOneByOne              3000  avgt   10  59.226 ? 14.135  ms/op
RocksDbBenchmark.persistOneByOne              10  avgt   10   0.016 ?  0.002  ms/op
RocksDbBenchmark.persistOneByOne             100  avgt   10   0.228 ?  0.273  ms/op
RocksDbBenchmark.persistOneByOne             500  avgt   10   2.285 ?  2.947  ms/op
RocksDbBenchmark.persistOneByOne            1000  avgt   10   8.565 ?  7.070  ms/op
RocksDbBenchmark.persistOneByOne            3000  avgt   10  37.588 ?  8.298  ms/op
```

This tests how many milliseconds on average it took 1 operation to complete.
10 operations were averaged out after warm up.

`?` is supposed to be `±`

This looks close enough to the results from #924 (since many factors can tilt the results each time)

Tests were run on Hetzner Cx21 machine running Ubuntu 18.04 LTS

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java [hunk 1]
```diff
@@ -334,10 +334,18 @@ public void deleteBatch(Collection<Pair<Indexable, ? extends Class<? extends Per
                     Indexable indexable = entry.low;
                     byte[] keyBytes = indexable.bytes();
                     ColumnFamilyHandle handle = classTreeMap.get(entry.hi);
-                    writeBatch.remove(handle, keyBytes);
+                    try {
+                        writeBatch.delete(handle, keyBytes);
+                    } catch (RocksDBException e) {
+                        log.error("Could not delete handle: " + handle.getID());
+                    }
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: I believe that which ever service calls delete batch should catch the error and decide how to handle it.

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java [hunk 2]
```diff
@@ -334,10 +334,18 @@ public void deleteBatch(Collection<Pair<Indexable, ? extends Class<? extends Per
...
                     ColumnFamilyHandle metadataHandle = metadataReference.get(entry.hi);
                     if (metadataHandle != null) {
-                        writeBatch.remove(metadataHandle, keyBytes);
+                        try {
+                            writeBatch.delete(metadataHandle, keyBytes);
+                        } catch (RocksDBException e) {
+                            log.error("Could not delete metadataHandle: " + metadataHandle.getID());
```
Comment (GalRogozinski) [inline src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: ditto

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 1 above)
Comment (legacycode) [reply src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: I wrapped it in a RuntimeException.

File: src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java (hunk 2 above)
Comment (legacycode) [reply src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java]: I wrapped it in a RuntimeException.

**Comment by legacycode** [comment]
done. is codacy not running anymore?

**Pull request merged**